### PR TITLE
(DOCSP-29690) Update Quickstart tests

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

Update the Quickstart tests to make them adhere to best practices. Convert them to TypeScript. Mostly leave the Quickstart content as-is. I didn't want to make this PR a total Quickstart overhaul.

### Jira

- https://jira.mongodb.org/browse/DOCSP-29690

### Staged Changes

- [Quickstart](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-29690-quickstart/sdk/node/quick-start/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/node/Examples/quickstart.js [hunk 1]
```diff
@@ -1,222 +1,205 @@
...
-      // :remove-start:
-      // wait 1 second until the collection listener has registered the modification and deletion events
-      setTimeout(() => {
-        expect(taskHasBeenModified).toBe(true);
-        expect(taskHasBeenDeleted).toBe(true);
-      }, 1000);
-      realm.write(() => {
-        realm.delete(task2); // task1 has already been deleted
+      // Update UI in response to modified objects.
+      // `newModifications` contains object indexes from after they were modified.
```
Comment (cbush) [inline examples/node/Examples/quickstart.js:81]: This is a bit unclear... what are we trying to say here?

I think the idea is that the index corresponds to the modified object's position in the collection after all deletions/insertions have been applied. (...which is why order of operations matters)

File: examples/node/Examples/quickstart.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+      // Update UI in response to inserted objects.
+      changes.insertions.forEach((index) => {
+        const insertedTasks = tasks[index];
+        console.log(`insertedTasks: ${JSON.stringify(insertedTasks, null, 2)}`);
+        // ...
+      });
+
+      // Update UI in response to modified objects.
+      // `newModifications` contains object indexes from after they were modified.
```
Comment (cbush) [inline examples/node/Examples/quickstart.ts:93]: Same comments apply here

**Review by cbush — APPROVED** [review]
Looks good!

File: examples/node/Examples/quickstart.js [hunk 3]
```diff
@@ -1,222 +1,205 @@
...
       // :remove-start:
-      expect(task1.status).toBe("InProgress");
+      if (changes.newModifications.length > 0) {
+        taskHasBeenModified = true;
+      }
+      if (changes.deletions.length > 0) {
+        taskHasBeenDeleted = true;
+      }
       // :remove-end:
+      // Update UI in response to deleted objects.
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply examples/node/Examples/quickstart.js:65]: Good point! I'll add this as an important admonition in the Quick Start itself.

File: examples/node/Examples/quickstart.js (hunk 1 above)
Comment (krollins-mdb) [reply examples/node/Examples/quickstart.js:81]: This was the original wording. I debated altering it and opted not to. But I think it is unclear, so I've updated it now.

**Pull request merged**